src/components/RulesModal.tsx

```
@@ -0,0 +1,181 @@
+      .replace(/\[(.+?)\]\((.+?)\)/g, '<a href="$2" target="_blank" rel="noopener noreferrer">$1</a>')
+      .replace(/&nbsp;/g, '\u00A0');
+  };
+
+  for (const line of lines) {
+    const trimmed = line.trim();
+
+    // Headers
+    if (trimmed.startsWith('## ')) {
+      flushList();
+      elements.push(
+        <h2 key={`h2-${keyCounter++}`} dangerouslySetInnerHTML={{ __html: formatInline(trimmed.slice(3)) }} />
```

Biome (2.4.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

(lint/security/noDangerouslySetInnerHtml)